src/posts/interning-at-khan-academy.md

```diff
@@ -0,0 +1,53 @@
+
+![changes to the math keypad](/images/interning-at-khan-academy/image_0.png)
+
+*an early screenshot of changes to the math keypad*
+
+# Part 2: iframes, iframes, iframes
+
+Now that I was used to the flow of picking a task, creating a code review, and deploying my changes, I started on my first project -- improving our content editor to be mobile first. This involved pulling together specs for what types of devices students use Khan Academy on, rendering content with the new styles the rest of the team was working on, isolating the rendered content to prevent styles leaking from the rest of the editor, and implementing touch emulation for interacting with the mobile previews.
+
+To implement the isolation of content, I decided to render previews inside of iframes. In addition to providing style isolation, iframes also made it possible to properly render components that are rendered directly to the body, such as the math keypad I worked on earlier. To transfer data between the parent editor and the child iframe, the parent first stores a JS object representation of the exercise content in a global variable, because the exercise content contains callback functions which prevent it from being serialized for window messaging. The parent then sends a message ([http://ejohn.org/blog/cross-window-messaging/](http://ejohn.org/blog/cross-window-messaging/)) to the iframe, notifying it of the updated content.
+
+When the iframe receives this message, it grabs the global JS object from the parent (because the iframe and parent are on the same domain this is allowed by the browser) and renders that content. With this implementation, preview updates happen immediately after content edits. In addition, with a special `shouldComponentUpdate` that only allows iframe reloads when they are needed for changes such as touch emulation, iframe loads are minimized so that in most cases the only loading delay happens during the initial editor load.
+
+![interactive mobile previews](/images/interning-at-khan-academy/image_1.png)
+
+*live-updated interactive mobile previews with iframes!*
+
+After these changes were implemented, I got the chance to work with content creators to test out my changes and make sure they didn’t interrupt their creation process. By working with content creators, I was able to understand exactly what they were hoping to get out of my changes, and I was able to tune the goals of my project to these needs to make sure that it was helping the content creators as much as possible. With all the kinks worked out, I was able to finally deploy the content editor that is now used by content creators every day! Even as I continued on to my next project, content creators continued to provide feedback that was worked through and helped further refine the editor.
```

Comment:
Some very brief examples of change requests, or at least the class of changes that they were interested in, would be valuable here :)